### PR TITLE
Module augmentation

### LINKED ISSUE #187: Document module augmentation
It should go on the declaration merging page.

 I believe this is documented on "What's New in TypeScript 1.8" and nowhere else.

### PR DESCRIPTION
**Pull request by sandersn**
Fixes #187 

Based on What's New in TypeScript 1.8

### CODE REVIEW

File: pages/Declaration Merging.md [hunk 1]
```diff
@@ -208,5 +207,70 @@ namespace Color {
...
+
+```js
+// observable.js
+export class Observable<T> {
+    // ... implementation left as an exercise for the reader ...
+}
+
+// map.js
+import { Observable } from "./observable";
+Observable.prototype.map = // ... another exercise for the reader
```
Comment (mhegazy) [inline pages/Declaration Merging.md]:
```js
function () {
 // ... another exercise for the reader
}
```

File: pages/Declaration Merging.md [hunk 2]
```diff
@@ -208,5 +207,70 @@ namespace Color {
...
+
+
+// consumer.ts
+import { Observable } from "./observable";
+import "./map";
+let o: Observable<number>;
+o.map(x => x.toFixed());
+```
+
+The module name is resolved the same way as module specifiers in `import`/`export`.
```
Comment (mhegazy) [inline pages/Declaration Merging.md]: we can add a link here to the module section.

**Comment by mhegazy** [comment]
:+1:

File: pages/Declaration Merging.md (hunk 1 above)
Comment (sandersn) [inline pages/Declaration Merging.md]: done

File: pages/Declaration Merging.md (hunk 2 above)
Comment (sandersn) [inline pages/Declaration Merging.md]: done

**Pull request merged**